controllers/horizon_controller.go

```diff
@@ -308,6 +313,36 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 
 	// run check OpenStack secret - end
 
+	// Create Memcached instance if no shared instance exists.
+	if instance.Spec.SharedMemcached == "" {
+		memcached := r.renderMemcached(instance)
+		op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), memcached, func() error {
+			memcached.Labels = map[string]string{"service": instance.Name}
+			memcached.Spec.Replicas = instance.Spec.Replicas
+
+			err := controllerutil.SetControllerReference(helper.GetBeforeObject(), memcached, helper.GetScheme())
+			if err != nil {
+				return err
+			}
+
+			return nil
+		})
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				r.Log.Info(fmt.Sprintf("Memcached %s not found", memcached.Name))
+				return ctrl.Result{RequeueAfter: time.Duration(5) * time.Second}, nil
+			}
+			return ctrl.Result{}, err
+		}
+		if op != controllerutil.OperationResultNone {
+			r.Log.Info(fmt.Sprintf("Memcached %s - %s", memcached.Name, op))
+		}
+
+		// Mark the Memcached Service as Ready if we get to this point with no errors
+		instance.Status.Conditions.MarkTrue(
+			horizonv1alpha1.HorizonMemcachedReadyCondition, horizonv1alpha1.HorizonMemcachedReadyMessage)
+	}
```

Comment:
(Can be follow-up)
Should we ensure we delete the default memcached CR in case SharredMemcache id given ?